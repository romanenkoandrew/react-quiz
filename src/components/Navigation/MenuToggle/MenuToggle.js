import React from "react";
import styles from "./MenuToggle.module.css";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faBars, faTimes } from "@fortawesome/free-solid-svg-icons";

const MenuToggle = (props) => {
  let isOpenStyle;
  if (props.isOpen) {
    isOpenStyle = "open";
  }
  return (
    <FontAwesomeIcon
      icon={props.isOpen ? faTimes : faBars}
      className={`${styles.MenuToggle} ${styles[isOpenStyle]}`}
      onClick={props.onToggle}
    />
  );
};

export default MenuToggle;
