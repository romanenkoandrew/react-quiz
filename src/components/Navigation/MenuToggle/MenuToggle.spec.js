import React from "react";
import { shallow } from "enzyme";
import MenuToggle from "./MenuToggle";

describe("MenuToggle component", () => {
  it("should render MenuToggle with props.isOpen", () => {
    const component = shallow(<MenuToggle isOpen={true} />);
    expect(component).toMatchSnapshot();
  });
  it("should render MenuToggle without props.isOpen", () => {
    const component = shallow(<MenuToggle isOpen={false} />);
    expect(component).toMatchSnapshot();
  });
});
