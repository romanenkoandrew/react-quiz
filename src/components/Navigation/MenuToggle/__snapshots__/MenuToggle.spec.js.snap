// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuToggle component should render MenuToggle with props.isOpen 1`] = `
<FontAwesomeIcon
  border={false}
  className="MenuToggle open"
  fixedWidth={false}
  flip={null}
  icon={
    Object {
      "icon": Array [
        352,
        512,
        Array [],
        "f00d",
        "M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z",
      ],
      "iconName": "times",
      "prefix": "fas",
    }
  }
  inverse={false}
  listItem={false}
  mask={null}
  pull={null}
  pulse={false}
  rotation={null}
  size={null}
  spin={false}
  swapOpacity={false}
  symbol={false}
  title=""
  transform={null}
/>
`;

exports[`MenuToggle component should render MenuToggle without props.isOpen 1`] = `
<FontAwesomeIcon
  border={false}
  className="MenuToggle undefined"
  fixedWidth={false}
  flip={null}
  icon={
    Object {
      "icon": Array [
        448,
        512,
        Array [],
        "f0c9",
        "M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z",
      ],
      "iconName": "bars",
      "prefix": "fas",
    }
  }
  inverse={false}
  listItem={false}
  mask={null}
  pull={null}
  pulse={false}
  rotation={null}
  size={null}
  spin={false}
  swapOpacity={false}
  symbol={false}
  title=""
  transform={null}
/>
`;
